firmware/cmake/OsccConfig.cmake

```diff
@@ -1,9 +1,35 @@
-if(KIA_SOUL)
-    add_definitions(-DKIA_SOUL)
+#To be deprecated in favor of using VEHICLE and VEHICLE VALUES so that
+#adding new vehicles can be maintained in a single list and the list of
+#available values can be printed in help and output.
+set(KIA_SOUL OFF CACHE BOOL "Build firmware for the petrol Kia Soul")
+set(KIA_SOUL_EV OFF CACHE BOOL "Build firmware for the Kia Soul EV")
+set(KIA_NIRO OFF CACHE BOOL "Build firmware for the Kia Niro")
+
+set(VEHICLE "" CACHE STRING
+  "VEHICLE chosen by the user at CMake configure time")
+
+set(VEHICLE_VALUES "kia_soul;kia_soul_ev;kia_niro" CACHE STRING
+  "List of possible values for the VEHICLE cache variable")
+
+set_property(CACHE VEHICLE PROPERTY STRINGS ${VEHICLE_VALUES})
+
+message(STATUS "VEHICLE='${VEHICLE}'")
+
+if (";${VEHICLE_VALUES};" MATCHES ";${VEHICLE};")
+  string(TOUPPER ${VEHICLE} VEHICLE_UPPER_CASE)
+  add_definitions(-D${VEHICLE_UPPER_CASE})
+elseif(KIA_SOUL)
+  add_definitions(-DKIA_SOUL)
+  set(VEHICLE kia_soul)
+  message(WARNING "KIA_SOUL is being deprecated for VEHICLE string assign '-DVEHICLE=kia_soul'")
```

Comment:
This (and below) would be clearer with the command spelled out, `'-DKIA_SOUL=ON' is being deprecated...`